caffe2/core/dispatch/TensorTypeId.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "caffe2/utils/IdWrapper.h"
+#include <string>
+#include <iostream>
+#include <mutex>
+#include <unordered_set>
+
+namespace c10 {
+
+namespace details {
+  using _tensorTypeId_underlyingType = uint8_t;
+}
+
+/**
+ * Dynamic type ID of a Tensor argument.  It represents something like CPUFloatTensor, etc.
```

Comment:
so according to c10 prototype it actually represents just CPUTensor, right? And the dtype is incorporated inside (and btw, I agree that it's the right functionality)

In that case the comment is not very representative.

For dispatch purposes - how are you going to handle the separate dtype - construct custom key, right?